x-pack/plugins/search_playground/public/components/search_mode/search_mode.tsx

```diff
@@ -8,18 +8,53 @@
 import {
   EuiButton,
   EuiEmptyPrompt,
+  EuiFieldText,
   EuiFlexGroup,
   EuiFlexItem,
-  EuiSearchBar,
+  EuiForm,
   useEuiTheme,
 } from '@elastic/eui';
-import React from 'react';
+import React, { useCallback } from 'react';
 import { css } from '@emotion/react';
+import { Controller, useController, useFormContext } from 'react-hook-form';
+import { i18n } from '@kbn/i18n';
+import { DEFAULT_PAGINATION } from '../../../common';
 import { ResultList } from './result_list';
+import { ChatForm, ChatFormFields } from '../../types';
+import { useSearchPreview } from '../../hooks/use_search_preview';
+import { getPaginationFromPage } from '../../utils/pagination_helper';
 
 export const SearchMode: React.FC = () => {
+  const {
+    data: { results, pagination, isInitialState },
+    fetchSearchResults,
+  } = useSearchPreview();
   const { euiTheme } = useEuiTheme();
-  const showResults = true; // TODO demo
+  const { control, handleSubmit } = useFormContext();
+  const {
+    field: { onChange: searchBarOnChange, value: searchBarValue },
+    formState: { isSubmitting },
+  } = useController<ChatForm, ChatFormFields.searchQuery>({
+    name: ChatFormFields.searchQuery,
+  });
+
+  const updateSearchQuery = useCallback(
+    (query: string) => {
+      searchBarOnChange(query);
+    },
+    [searchBarOnChange]
+  );
+  const handleSearch = async (query = searchBarValue, paginationParam = DEFAULT_PAGINATION) => {
+    try {
+      await fetchSearchResults({ query, pagination: paginationParam });
+    } catch (e) {
+      // TODO handle error ?
```

Comment:
We probably want to log the error. What do you think?

Also, Should we display a pop-up or alert of some type and display and entry page on the Results View?

Reply:
Not a popup but rather a toast, or something on the screen. I will try to come up with something similar to Stack implementation.